### PR TITLE
Example of how to optimize specific databases for testing

### PR DESCRIPTION
**Pull request by goldbergyoni**

### CODE REVIEW

File: various-receipes/db-optimization/docker-compose-mysql.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+version: '3.6'
+services:
+  database:
+    image: mysql:8
+    command: --default-authentication-plugin=mysql_native_password --innodb_flush_log_at_trx_commit=2 --sync-binlog=0
```
Comment (goldbergyoni) [inline various-receipes/db-optimization/docker-compose-mysql.yml:5]: --innodb_flush_log_at_trx_commit=0 seems faster (not even writing immediately to log)

sync-binlog - Why not greater value like 20?

- Why not disable logging with --log-level=0? 

This comments applies to all DB 👆

File: various-receipes/db-optimization/docker-compose-mysql.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    environment:
+      - MYSQL_ROOT_PASSWORD=example
+      - MYSQL_USER=myuser
+      - MYSQL_PASSWORD=myuserpassword
+      - MYSQL_DATABASE=shop
+    container_name: 'mysql-for-testing'
+    ports:
+      - '33060:3306'
+    # Do not use Memory engine, use RAM folder (tmpfs) instead (https://github.com/testjavascript/nodejs-integration-tests-best-practices/issues/9#issuecomment-710674437)
+    tmpfs: /var/lib/mysql
```
Comment (goldbergyoni) [inline various-receipes/db-optimization/docker-compose-mysql.yml:15]: Also map the data log to tmpfs, many IO goes to DB log (not logger rather the data log)?

This comments applies to all DB 👆

File: various-receipes/db-optimization/docker-compose-mongo.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+version: '3.6'
+services:
+  database:
+    image: mongo:4
```
Comment (goldbergyoni) [inline various-receipes/db-optimization/docker-compose-mongo.yml:4]: No optimizations? I know it has a built-in in-mem engine, may be only commercial

File: various-receipes/db-optimization/docker-compose-postgres.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+version: '3.6'
+services:
+  database:
+    image: postgres:11
+    command: postgres -c fsync=off -c synchronous_commit=off -c full_page_writes=off -c random_page_cost=1.0
```
Comment (goldbergyoni) [inline various-receipes/db-optimization/docker-compose-postgres.yml:5]: My PR shows how to disable logging also

**Review by goldbergyoni — COMMENTED** [review]
@mikicho I left comments and anyway can't approve since I'm the opener

p.s. If you search for the word 'test' here, you'll find more examples for MySQL flags:

https://dev.mysql.com/doc/refman/8.0/en/innodb-parameters.html#sysvar_innodb_fsync_threshold

File: various-receipes/db-optimization/docker-compose-mysql.yml (hunk 1 above)
Comment (mikicho) [reply various-receipes/db-optimization/docker-compose-mysql.yml:5]: > --innodb_flush_log_at_trx_commit=0 seems faster (not even writing immediately to log)

Seems like you're correct

> sync-binlog - Why not greater value like 20?

0 disable the feature which is probably faster.

> Why not disable logging with --log-level=0

I think we already do so for pg, couldn't find something like this for MySQL

File: various-receipes/db-optimization/docker-compose-mysql.yml (hunk 2 above)
Comment (mikicho) [reply various-receipes/db-optimization/docker-compose-mysql.yml:15]: > By default no log files are enabled in MYSQL.

https://stackoverflow.com/a/29409969/3266867

Comment (goldbergyoni) [reply various-receipes/db-optimization/docker-compose-mysql.yml:15]: @mikicho This suggestion is not related with observability logging rather write-ahead logging (WAL)

File: various-receipes/db-optimization/docker-compose-postgres.yml (hunk 4 above)
Comment (mikicho) [reply various-receipes/db-optimization/docker-compose-postgres.yml:5]: what do you mean? I didn't change anything in the Postgres docker

File: various-receipes/db-optimization/docker-compose-mongo.yml (hunk 3 above)
Comment (mikicho) [reply various-receipes/db-optimization/docker-compose-mongo.yml:4]: Couldn't find any CLI option for tuning, do you know some?
The in-mem engine is enterprise-only AFAIK

**Comment by mikicho** [comment]
> p.s. If you search for the word 'test' here, you'll find more examples for MySQL flags:
https://dev.mysql.com/doc/refman/8.0/en/innodb-parameters.html#sysvar_innodb_fsync_threshold

I think we have enough flags for now, I'm sure the community will help us tune it even further :)

**Comment by goldbergyoni** [comment]
> > p.s. If you search for the word 'test' here, you'll find more examples for MySQL flags:
> > https://dev.mysql.com/doc/refman/8.0/en/innodb-parameters.html#sysvar_innodb_fsync_threshold
> 
> I think we have enough flags, for now, I'm sure the community will help us tune it even further :)

@mikicho Agree 💪

File: various-receipes/db-optimization/docker-compose-mongo.yml (hunk 3 above)
Comment (rluvaton) [reply various-receipes/db-optimization/docker-compose-mongo.yml]: @goldbergyoni I looked how [MongoDB Memory Server NPM package](https://www.npmjs.com/package/mongodb-memory-server) been implemented and I saw that he use the storage engine `ephemeralForTest` which been said in [this SO answer](https://stackoverflow.com/a/67110672/5923666) that:
> Note that `ephemeralForTest` storage engine doesn't support transactions and is therefore of limited use on MongoDB 4.0 and newer servers.

although there seems to be no support 

so after some searching I found that what you do is what has been suggested [here](https://jira.mongodb.org/browse/SERVER-38762?focusedCommentId=3204204&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3204204)

Comment (mikicho) [reply various-receipes/db-optimization/docker-compose-mongo.yml]: Thanks @rlucaton !

**Pull request merged**